compiler+runtime/include/cpp/jank/runtime/obj/delay.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+namespace jank::runtime
+{
+  template <>
+  struct static_object<object_type::delay> : gc
+  {
+    static constexpr native_bool pointer_free{ false };
+
+    static_object() = default;
+    static_object(object_ptr o);
```

Comment:
CI will complain that the arg name `o` in the declaration doesn't match the arg name `fn` in the definition. We want to update the declaration, since `fn` is much more helpful.